### PR TITLE
Add support for sliderwhistle

### LINKED ISSUE #12325: "sliderslide" hitsamples do not seem to be playing
As per title. Using class skin.

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes https://github.com/ppy/osu/issues/12325

Turns out `sliderwhistle` samples were just not used at all. Confirmed to fix via disco prince and the map mentioned by the twitter user in the above issue.

This `Sample` overriding stuff has to go from `Slider`, but that's a task for another day I think - maybe it's time to place a judgement-less hitobject at the correct time for playing the tail samples.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 1]
```diff
@@ -81,6 +81,9 @@ public override Vector2 Position
 
         public List<IList<HitSampleInfo>> NodeSamples { get; set; } = new List<IList<HitSampleInfo>>();
 
+        [JsonIgnore]
+        public IList<HitSampleInfo> OriginalSamples { get; private set; }
```
Comment (frenzibyte) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs:85]: This is not updating on changes to `Samples` / `SamplesBindable`, i.e. in editor, adding or removing the whisper sample from the slider will not update the drawable object.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/Slider.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs:85]: Oh I thought that such changes would call `ApplyDefaults`? Will need to check but I’m not sure it’s correct right now either.

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs]: Fixed with the newest changes.

**Comment by smoogipoo** [comment]
I ended up removing the overwriting of `Samples` completely, and storing them as a separate member inside `Slider`. Still probably needs to be implemented as a nested hitobject at some point but this should do for now.

This also brought to light the fact that the legacy encoder wasn't writing samples correctly, which has also been fixed and the relevant hack removed from the encoder.

**Review by peppy — APPROVED** [review]

**Pull request merged**